sky/backends/backend_utils.py

```diff
@@ -1827,26 +1837,30 @@ def _query_cluster_status_via_cloud_api(
         cloud_name = repr(handle.launched_resources.cloud)
         try:
             node_status_dict = provision_lib.query_instances(
```

Comment:
It may be pretty easy for new code/contributors to call provision_lib funcs and pass `cluster_name=cluster_name`. Hard to catch if we don't rename the args in provision_lib interface. Is this too intrusive? Any ideas to mitigate this?

One possibility/bandage is to document this clearly at the top of provision_lib.

Reply:
Another potential way is to create a lightweight class and use it in provision_lib interfaces:
```
class ClusterIdentity:  # or better name
  cluster_name: str
  cluster_name_on_cloud: str
  cloud: str 
```

Reply:
This is a great point! I could not think of a better way to mitigate this. I would personally prefer renaming the `cluster_name` in `provision_lib` to `cluster_name_on_cloud`, and add a comment at the top of `provision_lib`.

Another alternative (though more involved) is to rename the `handle.cluster_name` to `handle.display_name` and rename `handle.cluster_name_on_cloud` to `handle.cluster_name`.

Reply:
After offline discussion, we now changed the `provision_lib`'s APIs to take `cluster_name_on_cloud` as an argument.